### PR TITLE
SmartOS imgadm module and state did not like docker images

### PR DESCRIPTION
**Pull request by sjorge**
### What does this PR do?
This PR fixes the importing and managing of docker images on SmartOS.

### What issues does this PR fix or reference?
N/a

### Previous Behavior
Exceptions were encountered when trying to import docker images.

### New Behavior
Docker images can now be imported and managed properly.

### Tests written?
No

### Commits signed with GPG?
No

### Backporting
It's a rather large fix so I would prefer to keep it in develop only, although there is no functional change compared to 2018.3.x so it may be worth while pulling this in.

### CODE REVIEW

**Comment by sjorge** [comment]
I will be super busy the coming week/weekend, but smartos_vmadm should also be updated to not error on docker images. Fixing this is a requirement for smartos_vmadm, I will deal with smartos_vmadm once this is merged and I am less busy.

**Comment by sjorge** [comment]
Lint and test failures seem to be unrelated.

File: salt/modules/smartos_imgadm.py [hunk 1]
```diff
@@ -150,25 +232,33 @@ def list_installed(verbose=False):
     verbose : boolean (False)
         toggle verbose output
 
+    .. versionchanged:: 2016.3.2
```
Comment (rallytime) [inline salt/modules/smartos_imgadm.py]: This should be `Fluorine`, yes?

File: salt/modules/smartos_imgadm.py [hunk 2]
```diff
@@ -89,6 +153,22 @@ def version():
     return ret[-1]
 
 
+def docker_to_uuid(uuid):
+    '''
+    Get the image uuid from an imported docker image
+
+    .. versionadded:: Neon
```
Comment (rallytime) [inline salt/modules/smartos_imgadm.py]: This should be Fluorine.

File: salt/modules/smartos_imgadm.py [hunk 3]
```diff
@@ -308,4 +415,91 @@ def vacuum(verbose=False):
         return list(result.keys())
 
 
+def sources(verbose=False):
+    '''
+    Return a list of available sources
+
```
Comment (rallytime) [inline salt/modules/smartos_imgadm.py]: This new function, as well as all of the the other new functions below, should have a versionadded tag for `Fluorine`. :)

**Review by rallytime — CHANGES_REQUESTED** [review]
Some minor documentation requests from me. :)

File: salt/modules/smartos_imgadm.py (hunk 1 above)
Comment (sjorge) [reply salt/modules/smartos_imgadm.py]: Done

File: salt/modules/smartos_imgadm.py (hunk 2 above)
Comment (sjorge) [reply salt/modules/smartos_imgadm.py]: Done

File: salt/modules/smartos_imgadm.py (hunk 3 above)
Comment (sjorge) [reply salt/modules/smartos_imgadm.py]: Done

**Comment by sjorge** [comment]
Done and rebased on develop

**Review by cachedout — APPROVED** [review]
Very nice.

**Review by rallytime — APPROVED** [review]
👍

**Pull request merged**